nixos/modules/services/development/jupyter/default.nix

```diff
@@ -2,29 +2,38 @@
   config,
   lib,
   pkgs,
+  options,
   ...
 }:
 let
 
   cfg = config.services.jupyter;
 
-  package = cfg.package;
+  package = pkgs.python3.withPackages (
+    ps:
+    [
+      cfg.package
+    ]
+    ++ cfg.extraPackages
+  );
```

Comment:
This is also a breaking change that should be mentioned in the changelog. This breaks any setup where `package = <some python environment>` as opposed to `package = <python package>` and using `extraPackages`.